code/game/objects/items/explosives/mine.dm

```diff
@@ -286,3 +286,109 @@
 	customizable = TRUE
 	matter = list("metal" = 3750)
 	has_blast_wave_dampener = TRUE
+
+
+
+/obj/item/explosive/mine/bounding
+	name = "\improper M5 Bounding Mine"
+	desc = "The M5 Bounding Mine is a landmine that propels itself upwards when the tripwire is set off. It explodes mid-air and either maims, kills, or fucks up anything standing in the blast zone. The 'Bouncing Betty' has seen limited use since the year 2150 due to the IFF sensors on the earlier models were prone to failure, resulting in friendly casualties. Be careful to not trip this or you're going to be ending up in a hospital bed full of shrapnel."
+	icon = 'icons/obj/items/weapons/grenade.dmi'
+	icon_state = "m5"
+	max_container_volume = 120
+	reaction_limits = list(	"max_ex_power" = 175,	"base_ex_falloff" = 75,	"max_ex_shards" = 32,
+							"max_fire_rad" = 5,		"max_fire_int" = 20,	"max_fire_dur" = 24,
+							"min_fire_rad" = 2,		"min_fire_int" = 3,		"min_fire_dur" = 3
+	)
+	angle = 360
+	use_dir = FALSE
+
+
+//arming
+/obj/item/explosive/mine/bounding/attack_self(mob/living/user)
+	if(!..())
+		return
+
+	if(check_for_obstacles(user))
+		return
+
+	if(active || user.action_busy)
+		return
+
+	user.visible_message(SPAN_NOTICE("[user] starts deploying [src]."), \
+		SPAN_NOTICE("You start deploying [src]."))
+	if(!do_after(user, 40, INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE))
+		user.visible_message(SPAN_NOTICE("[user] stops deploying [src]."), \
+			SPAN_NOTICE("You stop deploying \the [src]."))
+		return
+
+	if(active)
+		return
+
+	if(check_for_obstacles(user))
+		return
+
+	user.visible_message(SPAN_NOTICE("[user] finishes deploying [src]."), \
+		SPAN_NOTICE("You finish deploying [src]."))
+
+	if(!hard_iff_lock)
+		iff_signal = user.faction
+
+	cause_data = create_cause_data(initial(name), user)
+	anchored = TRUE
+	playsound(loc, 'sound/weapons/mine_armed.ogg', 25, 1)
```

Comment:
Please use named arguments here for ambiguous parameters (ones that aren't obvious from a glance).
For example, what is the 1 supposed to mean? What does the 25 mean? For someone who's been programming on this repo for a while, it'd be obvious, but for someone who hasn't, it's very confusing. 
Do not use integers for booleans. Use TRUE/FALSE
```suggestion
	playsound(loc, 'sound/weapons/mine_armed.ogg', volume = 25, vary = TRUE)
```
Do this for the rest of the procs you use.

Reply:
Sorry to say but all I did here was copy and paste what was previously up there, and thus this code has actually been in here for years. this is literally my first pr of actually doing anything big and ambitious so I apologize if it's not up to your liking. I'll commit it anyway because less shitcode the better I tell ya what